packages/server/src/workers/download.ts

```diff
@@ -104,13 +110,22 @@ export async function addDownloadJobs(resource: WithId<Resource>, context: Backg
 
   const ctx = tryGetRequestContext();
   for (const attachment of getAttachments(resource)) {
-    if (!isExternalUrl(attachment.url) || !isUrlAllowedByProject(project, attachment.url)) {
+    // Only process allowed external URLs
+    const url = attachment.value.url;
+    if (!isExternalUrl(url) || !isUrlAllowedByProject(project, url)) {
+      continue
+    }
+
+    // Skip if this mutation didn't adjust the URL in question
+    const pointer = Pointer.fromJSON(`${pathToPointer(attachment.path)}/url`);
+    if (pointer.get(resource) === pointer.get(previousVersion)) {
```

Comment:
There are some instances where this doesn't work as expected. For example, if you delete the first entry in an array, that's going to look to this code like all the remaining objects were mutated.

I _think_ that the outcome in that kind of scenario is mild enough not to warrant spending additional time fixing. It would mean that jobs for each remaining URL could be double-enqueued, which could result in creating extraneous Binary resources that have no references to them. This is probably worth calling out in a comment as a tradeoff we're explicitly making.